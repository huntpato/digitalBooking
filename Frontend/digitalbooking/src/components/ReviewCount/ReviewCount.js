import styles from "./review.module.css";

export const ReviewCount = (props) => {
  const { rate } = props;
  const { review } = styles;
  let rateDetail = " ";

  const rateDetails = (review) => {
    if(rate === null){
      rateDetail = "Nuevo";
    }else if (rate === 10) {
      rateDetail = "Excelente";
    } else if (rate < 10 && rate >= 8) {
      rateDetail = "Muy bueno";
    } else if (rate < 8 && rate >= 6) {
      rateDetail = "Bueno";
    } else if (rate < 6 && rate >= 4) {
      rateDetail = "Regular";
    } else if (rate < 4 && rate >= 0) {
      rateDetail = "Malo";
    } 
    return rateDetail;
  };

  rateDetails();

  return (
    <div className={review}>
      <h5>{rate != null ? rate : "*"}</h5>
      <h6>{rateDetail}</h6>
    </div>
  );
};
