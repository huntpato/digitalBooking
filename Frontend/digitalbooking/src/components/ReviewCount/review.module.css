.review{
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    margin-right: 0px;
    width: 20%;
}

.review h5{
    background-color: #383B58;
    border-radius: 6px;
    color: var(--background-white);
    font-size: 16px;
    margin-bottom: 2px;
    padding:4px 4px 4px 4px;
    text-align: center;
    width: 50%;
}

.review h6{
    align-self: stretch;
    font-size: 12px;
    text-align: right;
    width: 100%;
}

@media (max-width: 480px){
    .review{
        margin-top: 16px;
    }
}