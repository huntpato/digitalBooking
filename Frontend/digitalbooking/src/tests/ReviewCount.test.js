
import React from 'react';
import { shallow } from 'enzyme';
import { ReviewCount } from '../components/ReviewCount/ReviewCount'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <ReviewCounts />', ()=> {
  
    const wrapper= shallow(<ReviewCount rate={10}/>);

    test('Comprobar que <ReviewCounts /> renderiza correctamente.', () => {
        const wrapper= shallow(<ReviewCount rate={10}/>);
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const wrapper= shallow(<ReviewCount rate={10}/>);
        const div = wrapper.find('div');
        expect(div.hasClass('review')).toBe(true);
    })

    test('Verificar que el componente realiza correctamente el cálculo del score "Excelente"', () => {
        const wrapper= shallow(<ReviewCount rate={10}/>);
        const h6= wrapper.find('h6').text();
        expect( h6 ).toContain("Excelente");

    })

    test('Verificar que el componente realiza correctamente el cálculo del score "Muy bueno"', () => {
        const wrapper= shallow(<ReviewCount rate={9}/>);
        const h6= wrapper.find('h6').text();
        expect( h6 ).toContain("Muy bueno");
    })

    test('Verificar que el componente realiza correctamente el cálculo del score "Bueno"', () => {
        const wrapper= shallow(<ReviewCount rate={7}/>);
        const h6= wrapper.find('h6').text();
        expect( h6 ).toContain("Bueno");
    })

    test('Verificar que el componente realiza correctamente el cálculo del score "Regular"', () => {
        const wrapper= shallow(<ReviewCount rate={5}/>);
        const h6= wrapper.find('h6').text();
        expect( h6 ).toContain("Regular");
    })

    test('Verificar que el componente realiza correctamente el cálculo del score "Malo"', () => {
        const wrapper= shallow(<ReviewCount rate={3}/>);
        const h6= wrapper.find('h6').text();
        expect( h6 ).toContain("Malo");
    })

    test('Verificar que el componente realiza correctamente el cálculo del score "Nuevo"', () => {
        const wrapper= shallow(<ReviewCount rate={null}/>);
        const h6= wrapper.find('h6').text();
        expect( h6 ).toContain("Nuevo");
    })

})
