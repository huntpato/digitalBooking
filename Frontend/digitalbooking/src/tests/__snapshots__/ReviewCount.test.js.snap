// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <ReviewCounts /> Comprobar que <ReviewCounts /> renderiza correctamente. 1`] = `
<div
  className="review"
>
  <h5>
    10
  </h5>
  <h6>
    Excelente
  </h6>
</div>
`;
